seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/TaskExecutionService.java

```diff
@@ -455,6 +459,37 @@ private synchronized void updateMetricsContextInImap() {
         contextMap.putAll(finishedExecutionContexts);
         contextMap.putAll(executionContexts);
         try {
+            ClusterState clusterState =
+                    nodeEngine.getHazelcastInstance().getCluster().getClusterState();
+            if (clusterState != ClusterState.ACTIVE) {
+                logger.warning(
+                        String.format(
+                                "The cluster is not ready yet, cluster state [%s], looking forward to the next "
+                                        + "scheduling",
+                                clusterState));
+                return;
+            }
+            // Waiting for cluster startup to complete
+            if (waitClusterStarted == null) {
+                waitClusterStarted =
+                        new CountDownLatch(
+                                nodeEngine.getHazelcastInstance().getCluster().getMembers().size());
+                nodeEngine
+                        .getHazelcastInstance()
+                        .getLifecycleService()
+                        .addLifecycleListener(
+                                event -> {
+                                    if (event.getState() == LifecycleEvent.LifecycleState.STARTED) {
+                                        waitClusterStarted.countDown();
+                                    }
+                                });
+            }
+            if (waitClusterStarted.getCount() > 0) {
+                logger.warning(
+                        "The cluster is not ready yet, looking forward to the next scheduling");
+                return;
+            }
+
```

Comment:
1、"Why is it necessary for all nodes to be STARTED? Isn't it enough to just judge if the ClusterState is ACTIVE?"
2、"The getLifecycleService should obtain the LifecycleListener of the current instance, which should only receive the status change of the current node. In a cluster mode, shouldn't the waitClusterStarted.getCount() always be non-zero?"

<img width="667" alt="image" src="https://github.com/apache/incubator-seatunnel/assets/83933160/6a09f69b-450b-429c-a17f-ab6b7a9f9a75">

Reply:
> 1、"Why is it necessary for all nodes to be STARTED? Isn't it enough to just judge if the ClusterState is ACTIVE?" 2、"The getLifecycleService should obtain the LifecycleListener of the current instance, which should only receive the status change of the current node. In a cluster mode, shouldn't the waitClusterStarted.getCount() always be non-zero?"
> 
> <img alt="image" width="667" src="https://user-images.githubusercontent.com/83933160/237607445-6a09f69b-450b-429c-a17f-ab6b7a9f9a75.png">

@ic4y   Just judging that ClusterState=ACTIVE will still throw the same exception.  Let me reconsider how to solve it

Reply:
I've reviewed the code, and the exception might be thrown because the current Node hasn't finished initializing. Try using nodeEngine.getNode().getState().equals(NodeState.ACTIVE) and see how it goes.

Reply:
> I've reviewed the code, and the exception might be thrown because the current Node hasn't finished initializing. Try using nodeEngine.getNode().getState().equals(NodeState.ACTIVE) and see how it goes.

@ic4y I have determined the cause of the null pointer, let me verify it first

Reply:
NodeEngine needs to first execute the start method before executing getMap after initializing invoiceContext
![image](https://github.com/apache/incubator-seatunnel/assets/6446530/f294ea83-ef70-4c9e-afe5-d7a080aa7f80)

Reply:
@ic4y PTAL

Reply:
It seems fine, and it's not throwing any errors now, right?

Reply:
> It seems fine, and it's not throwing any errors now, right?

Yes, no exception was thrown. Through debugging, it has been clarified that NullPointerException is caused by the uninitialized InvocationContext